flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/RemoteInputChannel.java

```diff
@@ -211,6 +213,25 @@ void retriggerSubpartitionRequest(int subpartitionIndex) throws IOException {
 		return Optional.of(new BufferAndAvailability(next, moreAvailable, getSenderBacklog()));
 	}
 
+	@Override
+	public Collection<Buffer> getInflightingBuffers(long checkpointId) throws IOException {
+		synchronized (receivedBuffers) {
+			inflightingBuffers.clear();
```

Comment:
why do we need this to be a field if we are clearing it on each call?

Reply:
Actually it can also be constructed in every call, and my previous thought was only for avoiding creating it repeatedly for GC concerns.

Reply:
Once allocation per checkpoint would probably not matter, but for the PoC it also doesn't matter